lib/ansible/modules/systemd_service.py

```diff
@@ -484,17 +484,17 @@ def main():
             masked = out.strip() == "masked"
 
             if masked != module.params['masked']:
-                result['changed'] = True
```

Comment:
keep the changed set here, otherwise you break check mode reporting

Reply:
Ok.